lxd/api_cluster.go

```diff
@@ -608,6 +608,15 @@ func clusterPutJoin(d *Daemon, req api.ClusterPut) response.Response {
 		// Add the cluster flag from the agent
 		version.UserAgentFeatures([]string{"cluster"})
 
+		client, err = cluster.Connect(req.ClusterAddress, d.endpoints.NetworkCert(), false)
+		if err != nil {
+			return err
+		}
```

Comment:
Please add:

```
defer client.Close()
```

here just after checking for the error. Otherwise the underlying network connection will be leaked.

And while at it, please do the with the original ```client``` object, when it gets created earlier in the same function around line 377.

Reply:
When we added `client.Close()` calls at lines 378 and 616, we got the errors `lxd/api_cluster.go:378:15: client.Close undefined (type lxd.InstanceServer has no field or method Close)
lxd/api_cluster.go:616:15: client.Close undefined (type lxd.InstanceServer has no field or method Close)
` when running `make`, so we did not originally include it. I am not sure if this is an error with my local environment, but I will add these changes to the branch.

Reply:
Oh sorry, I think I confused this ```client``` with another one. There is no ```Close()``` method indeed. Please scratch that, and revert the branch to your original version, otherwise it won't compile. Apologies for the confusion.

Reply:
No worries, I reverted it just now.